elide-async/src/test/java/com/yahoo/elide/async/service/AsyncQueryThreadTest.java

```diff
@@ -0,0 +1,105 @@
+    private Elide elide;
+    private QueryRunner runner;
+    private AsyncQuery queryObj;
+    private AsyncQueryDAO asyncQueryDao;
+
+    @BeforeEach
+    public void setupMocks() {
+        user = mock(User.class);
+        elide = mock(Elide.class);
+        runner = mock(QueryRunner.class);
+        queryObj = mock(AsyncQuery.class);
+        asyncQueryDao = mock(DefaultAsyncQueryDAO.class);
+        queryThread = new AsyncQueryThread(queryObj, user, elide, runner, asyncQueryDao, "v1");
+    }
+
+    @Test
+    public void testAsyncQueryCleanerThreadSet() {
+        assertEquals(queryObj, queryThread.getQueryObj());
+        assertEquals(user, queryThread.getUser());
+        assertEquals(elide, queryThread.getElide());
+        assertEquals(runner, queryThread.getRunner());
+        assertEquals(asyncQueryDao, queryThread.getAsyncQueryDao());
+    }
+
+    @Test
+    public void testProcessQueryJsonApi() {
+        String query = "/group?sort=commonName&fields%5Bgroup%5D=commonName,description";
+        ElideResponse response = mock(ElideResponse.class);
+
+        when(queryObj.getQuery()).thenReturn(query);
+        when(queryObj.getQueryType()).thenReturn(QueryType.JSONAPI_V1_0);
+        when(elide.get(anyString(), any(), any(), anyString())).thenReturn(response);
+        when(response.getResponseCode()).thenReturn(200);
+        when(response.getBody()).thenReturn("ResponseBody");
+
+        queryThread.processQuery();
+
+        verify(asyncQueryDao, times(1)).updateStatus(queryObj, QueryStatus.PROCESSING);
+        verify(asyncQueryDao, times(1)).updateStatus(queryObj, QueryStatus.COMPLETE);
+        verify(asyncQueryDao, times(1)).createAsyncQueryResult(anyInt(), anyString(), any(), any());
+    }
+
+    @Test
+    public void testProcessQueryGraphQl() {
+        String query = "{\"query\":\"{ group { edges { node { name commonName description } } } }\",\"variables\":null}";
+        ElideResponse response = mock(ElideResponse.class);
+
+        when(queryObj.getQuery()).thenReturn(query);
+        when(queryObj.getQueryType()).thenReturn(QueryType.GRAPHQL_V1_0);
+        when(runner.run(query, user)).thenReturn(response);
+        when(response.getResponseCode()).thenReturn(200);
+        when(response.getBody()).thenReturn("ResponseBody");
+
+        queryThread.processQuery();
+
+        verify(asyncQueryDao, times(1)).updateStatus(queryObj, QueryStatus.PROCESSING);
+        verify(asyncQueryDao, times(1)).updateStatus(queryObj, QueryStatus.COMPLETE);
+        verify(asyncQueryDao, times(1)).createAsyncQueryResult(anyInt(), anyString(), any(), any());
+    }
+
+    @Test
+    public void testProcessQueryNoResponse() {
+        String query = "{\"query\":\"{ group { edges { node { name commonName description } } } }\",\"variables\":null}";
+
+        when(queryObj.getQuery()).thenReturn(query);
+        when(queryObj.getQueryType()).thenReturn(QueryType.GRAPHQL_V1_0);
+        when(runner.run(query, user)).thenReturn(null);
```

Comment:
In practice, this doesn't happen.  What does happen is that the queryRunner might throw an exception.  We should mock and test that scenario.